crates/sui-graphql-rpc/src/types/gas.rs

```diff
@@ -60,6 +60,11 @@ impl GasInput {
         last: Option<u64>,
         before: Option<object::Cursor>,
     ) -> Result<Connection<String, Object>> {
+        // A possible user error during dry run or execution would be to supply a gas payment that
+        // is not a Move object (i.e a package). Even though the transaction would fail to run, it
+        // will still be indexed. On loading the transaction, if the return type of this field is a
```

Comment:
nit: it won't be indexed, because this kind of transaction should fail during signing, but from the perspective of the GraphQL service that issued the execution/dry-run, it still needs to present the results.